File: hw/vaca/u3/HW18Singleton/mavenproject1/src/main/java/ec/edu/espe/mavenproject1/Calculator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 */

package ec.edu.espe.mavenproject1;

/**
 *
 * @author Kevin Vaca Edison's OOP ESPE
 */
public class Calculator {
    public static void main(String[] args) {
        // Get the USTax singleton instance
        USTax taxCalculator = USTax.getInstance();
        
        // Example sales amount
        double saleAmount = 100.0;
        
        // Display current tax percentage
        System.out.println("Current Tax Percentage: " + 
                          taxCalculator.getTaxPercentage() + "%");
        
        // Calculate total with default tax
        double totalWithTax = taxCalculator.salesTotal(saleAmount);
        System.out.println("Sale Amount: $" + saleAmount);
        System.out.println("Total with Default Tax: $" + totalWithTax);
        
        // Change tax percentage (if needed)
        taxCalculator.setTaxPercentage(8.5); // Set to 8.5%
        
        // Calculate with new tax rate
        totalWithTax = taxCalculator.salesTotal(saleAmount);
        System.out.println("\nUpdated Tax Percentage: " + 
                          taxCalculator.getTaxPercentage() + "%");
        System.out.println("Total with Updated Tax: $" + totalWithTax);
    }
}

File: hw/vaca/u3/SingletonJs/Calculator.js
document.addEventListener('DOMContentLoaded', () => {
    // Get the USTax singleton instance
    const taxCalculator = USTax.getInstance();
    
    // DOM elements
    const amountInput = document.getElementById('amount');
    const taxPercentageInput = document.getElementById('tax-percentage');
    const calculateBtn = document.getElementById('calculate');
    const resultDiv = document.getElementById('result');
    const currentTaxSpan = document.getElementById('current-tax');
    
    // Display current tax percentage
    currentTaxSpan.textContent = taxCalculator.getTaxPercentage();
    
    // Set up event listeners
    taxPercentageInput.addEventListener('change', (e) => {
        const newPercentage = parseFloat(e.target.value);
        if (!isNaN(newPercentage)) {
            taxCalculator.setTaxPercentage(newPercentage);
            currentTaxSpan.textContent = newPercentage;
        }
    });
    
    calculateBtn.addEventListener('click', () => {
        const amount = parseFloat(amountInput.value);
        if (!isNaN(amount)) {
            const total = taxCalculator.salesTotal(amount);
            resultDiv.innerHTML = `
                <p>Sale Amount: $${amount.toFixed(2)}</p>
                <p>Tax Rate: ${taxCalculator.getTaxPercentage()}%</p>
                <p>Total with Tax: $${total.toFixed(2)}</p>
            `;
        } else {
            resultDiv.textContent = 'Please enter a valid amount';
        }
    });
});